internal/controller/etcdcluster_controller.go

```diff
@@ -126,20 +142,36 @@ func (r *EtcdClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 // ensureClusterObjects creates or updates all objects owned by cluster CR
 func (r *EtcdClusterReconciler) ensureClusterObjects(
 	ctx context.Context, cluster *etcdaenixiov1alpha1.EtcdCluster) error {
+
 	if err := factory.CreateOrUpdateClusterStateConfigMap(ctx, cluster, r.Client); err != nil {
+		log.Error(ctx, err, "reconcile cluster state configmap failed")
 		return err
+	} else {
+		log.Debug(ctx, "cluster state configmap reconciled")
 	}
 	if err := factory.CreateOrUpdateHeadlessService(ctx, cluster, r.Client); err != nil {
+		log.Error(ctx, err, "reconcile headless service failed")
 		return err
+	} else {
+		log.Debug(ctx, "headless service reconciled")
 	}
 	if err := factory.CreateOrUpdateStatefulSet(ctx, cluster, r.Client); err != nil {
+		log.Error(ctx, err, "reconcile statefulset failed")
 		return err
+	} else {
+		log.Debug(ctx, "statefulset reconciled")
 	}
 	if err := factory.CreateOrUpdateClientService(ctx, cluster, r.Client); err != nil {
+		log.Error(ctx, err, "reconcile client service failed")
 		return err
+	} else {
+		log.Debug(ctx, "client service reconciled")
 	}
 	if err := factory.CreateOrUpdatePdb(ctx, cluster, r.Client); err != nil {
+		log.Error(ctx, err, "reconcile pdb failed")
 		return err
+	} else {
+		log.Debug(ctx, "pdb reconciled")
 	}
```

Comment:
I will go through the code and find more redundant elses and remove duplicated logs.

I removed them where I found, but will do it cleaner

Reply:
I looked into the code and found what you meant. I think that errors are not the same. Error that cluster objects couldn't be ensured is common one. Object reconciliation errors are detailed.

If we leave only common errors => it will not be clear for a customer where is the error exactly. If we remove common errors, then customer can lose the connection of too detailed errors. In my opinion chain of the errors should lead customer to the failure reason.

But at the same time I do not want to overload customer with errors. I think we can log less frequently, but anyway some error chains should exist (like a stack-trace). However I do not have long experience as a developer - these are just my engineering thoughts.

I removed common error and removed multiple redundant else's